### PR TITLE
[rfr] Update dockerfiles to support running registries locally [#OSF-7616]

### PR DESCRIPTION
**Pull request by abought**
## Purpose
Allow running registries from within docker compose for ongoing development.

This is a developer-facing change and is not expected to affect production environments. Users should not notice any difference.

## Changes
Update docker compose and local-dist settings examples with the changes required to get registries running locally. Uncomments and slightly updates code as needed.

Also fixes an issue running mfr sync locally.

## Side effects
None expected.

## Ticket
https://openscience.atlassian.net/browse/OSF-7616

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -216,21 +216,21 @@ services:
...
-  #     - 4300
-  #   ports:
-  #     - 4300:4300
-  #   volumes:
-  #     - registries_dist_vol:/code/dist
-  #   stdin_open: true
-  #   tty: true
+  registries:
+    image: quay.io/centerforopenscience/ember-osf-registries:develop-development-local
+    build: ../ember-osf-registries
```
Comment (alexschiller) [inline docker-compose.yml]: This line needs to be removed as it defaulted to my local ember-osf-registries instead of the image

Comment (abought) [inline docker-compose.yml]: cc @binoculars as I understand he's working on a similar PR.

Comment (binoculars) [inline docker-compose.yml]: Should be fixed after my PR to @hmoco's PR goes in, hmoco/osf.io#11 → #6952

You don't need the build property here. If anything, it should go in the override file. It's only to specify a different docker container (i.e. you wanted to update something in the registries Dockerfile)

**Comment by abought** [comment]
Pushed requested changes. Per discussion with @binoculars , we'll both keep our respective PRs open as there is not total overlap.

Thanks for the catch, Alex!

**Comment by brianjgeiger** [comment]
There are merge conflicts due to similar changes for EmberOSF. Could you fix?

**Comment by abought** [comment]
Thanks. I'm in a final round of manual testing, at which point I will push two more small changes discovered while integrating Barrett's changes.

**Comment by abought** [comment]
Sorry for the delay. I've pushed the changes; Alex has confirmed that the `volumes` change was helpful for him as well.

File: docker-compose.yml [hunk 2]
```diff
@@ -212,8 +212,6 @@ services:
     volumes:
       - registries_dist_vol:/code/dist
     stdin_open: true
-    tty: true
```
Comment (alexschiller) [inline docker-compose.yml]: 👍

File: docker-compose.override.yml [hunk 3]
```diff
@@ -1,9 +1,9 @@
 
 version: '2'
 
-volumes:
-  emberosf-sync:
-    external: true
+#volumes:
+#  emberosf-sync:
+#    external: true
```
Comment (alexschiller) [inline docker-compose.override.yml]: @brianjgeiger FYI: Anyone who pulled today will fail on `docker-compose up` until this change goes in.

**Pull request merged**